ui/src/app/pages/analysis/notebook-redirect.tsx

```diff
@@ -286,47 +325,34 @@ export const NotebookRedirect = fp.flow(withUserProfile(), withCurrentWorkspace(
             this.incrementProgress(Progress.Initializing);
           }
 
-          this.scheduleClusterPoll(billingProjectId);
+          this.clearAndSetTimeout(() => this.pollForRunningCluster(billingProjectId), clusterPollingIntervalMillis);
         }
       } catch (e) {
         if (!isAbortError(e)) {
-          this.scheduleClusterPoll(billingProjectId);
           throw e;
         }
       }
     }
 
-    private async getDefaultCluster(billingProjectId) {
-      const resp = await this.clusterRetry(() => clusterApi().listClusters(
-        billingProjectId, this.props.workspace.id, {signal: this.aborter.signal}));
-      return resp.defaultCluster;
-    }
-
-    private scheduleClusterPoll(billingProjectId) {
-      this.clearAndSetInterval(() => this.pollForRunningCluster(billingProjectId), clusterPollingIntervalMillis);
-    }
-
-    private async clusterRetry<T>(f: () => Promise<T>): Promise<T> {
-      return await retry(f, clusterApiRetryTimeoutMillis, clusterApiRetryAttempts);
-    }
-
     private async pollForRunningCluster(billingProjectId) {
       try {
         const cluster = await this.getDefaultCluster(billingProjectId);
         if (cluster.status === ClusterStatus.Running) {
-          clearInterval(this.intervalReference);
           await this.connectToRunningCluster(cluster);
         } else {
           // re-start cluster if stopped, and try again in the next polling interval
           if (cluster.status === ClusterStatus.Stopped) {
             await this.clusterRetry(() => notebooksClusterApi().startCluster(
               cluster.clusterNamespace, cluster.clusterName, {signal: this.aborter.signal}));
           }
+
+          // TODO(RW-3097): Backoff, or don't retry forever.
+          this.clearAndSetTimeout(() => this.pollForRunningCluster(billingProjectId), clusterPollingIntervalMillis);
         }
       } catch (e) {
         if (isAbortError(e)) {
           // stop polling on abort
```

Comment:
I'm guessing this was leftover from the interval, but this comment is no longer accurate and calling `clearTimeout` here is no longer necessary